iree/turbine/kernel/wave/runtime/CMakeLists.txt

```diff
@@ -22,13 +22,26 @@ find_package(nanobind CONFIG REQUIRED)
 
 # Link against torch, using the libraries and include dirs in site-packages.
 execute_process(
-  COMMAND "${Python_EXECUTABLE}" -c "import torch; print(torch.utils.cmake_prefix_path)"
-  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_cmake_PATH)
-list(APPEND CMAKE_PREFIX_PATH ${torch_cmake_PATH})
-list(APPEND CMAKE_PREFIX_PATH "/opt/rocm")
+  COMMAND "${Python_EXECUTABLE}" -c "import torch; print(torch.__path__[0])"
+  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_PATH)
 
-find_package(Torch REQUIRED)
-find_package(HIP REQUIRED)
+# Add all directories in /opt/rocm/lib/cmake to CMAKE_PREFIX_PATH
+file(GLOB subdirectories "/opt/rocm/lib/cmake/*")
```

Comment:
ROCm can be installed to other locations (especially coming from projects like https://github.com/ROCm/TheRock, or on Windows instead of Linux). There should be _zero_ hardcoded references to `/opt/rocm/`. The first preference should be the `ROCM_PATH` environment variable.

Reply:
thanks, that makes sense.  So how would I access ROCM_PATH? Several clusters on which I am running this do not have that environment variable set.

Reply:
The pattern I've seen used is to default to the environment variable then fall back to `/opt/rocm` on Linux if the environment variable is not set.

Reply:
Here are the official docs: https://rocm.docs.amd.com/en/latest/conceptual/cmake-packages.html . These recommend `find_package(hip REQUIRED)`. I don't think you can avoid a dependency on ROCm/HIP by relying on what torch provides and trusting that this path will exist, though the new PyTorch + ROCm wheels that @stellaraccident is putting together may provide some new options. Until then, it would be more robust to follow the recommendations on that page.